cpp/src/reductions/reductions.cpp

```diff
@@ -35,27 +35,31 @@ namespace detail {
 struct reduce_dispatch_functor {
   column_view const col;
   data_type output_dtype;
+  std::optional<std::reference_wrapper<const scalar>> init;
   rmm::mr::device_memory_resource* mr;
   rmm::cuda_stream_view stream;
 
   reduce_dispatch_functor(column_view const& col,
                           data_type output_dtype,
+                          std::optional<std::reference_wrapper<const scalar>> init,
                           rmm::cuda_stream_view stream,
                           rmm::mr::device_memory_resource* mr)
-    : col(col), output_dtype(output_dtype), mr(mr), stream(stream)
+    : col(col), output_dtype(output_dtype), init(init), mr(mr), stream(stream)
   {
   }
 
   template <aggregation::Kind k>
   std::unique_ptr<scalar> operator()(std::unique_ptr<reduce_aggregation> const& agg)
   {
     switch (k) {
-      case aggregation::SUM: return reduction::sum(col, output_dtype, stream, mr); break;
-      case aggregation::PRODUCT: return reduction::product(col, output_dtype, stream, mr); break;
-      case aggregation::MIN: return reduction::min(col, output_dtype, stream, mr); break;
-      case aggregation::MAX: return reduction::max(col, output_dtype, stream, mr); break;
-      case aggregation::ANY: return reduction::any(col, output_dtype, stream, mr); break;
-      case aggregation::ALL: return reduction::all(col, output_dtype, stream, mr); break;
+      case aggregation::SUM: return reduction::sum(col, output_dtype, init, stream, mr); break;
+      case aggregation::PRODUCT:
+        return reduction::product(col, output_dtype, init, stream, mr);
+        break;
```

Comment:
No break after return.
```suggestion
        return reduction::product(col, output_dtype, init, stream, mr);
```

Reply:
This also applies to other cases. No break after return. Otherwise the compiler may issue warnings.